core/sodasql/scan/dialect.py

```diff
@@ -109,9 +111,31 @@ def create_for_warehouse_type(cls, warehouse_type):
                     KEY_WAREHOUSE_TYPE: warehouse_type}))
 
     def default_connection_properties(self, params: dict):
-        # to be overriden by subclass
+        # to be overridden by subclass
+        pass
+
+    def safe_connection_data(self):
```

Comment:
Judging by the concrete implementations below, unless I'm mistaken, we end up sending usernames and passwords which I think doesn't reflect the name or docstring.